sql/sem/builtins: TestGenerateUniqueUnorderedIDOrder failed

sql/sem/builtins.TestGenerateUniqueUnorderedIDOrder [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11297752?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11297752?buildTab=artifacts#/) on master @ [6b0f43a13c4d2b36b4da5961e523161dabe212c0](https://github.com/cockroachdb/cockroach/commits/6b0f43a13c4d2b36b4da5961e523161dabe212c0):


```
=== RUN   TestGenerateUniqueUnorderedIDOrder
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/ad72371e92981b2b572c54a0422f4d25/logTestGenerateUniqueUnorderedIDOrder567054207
    test_log_scope.go:81: use -show-logs to present logs inline
    builtins_test.go:150: swap counts: []int{314, 291, 283, 269, 305, 299, 279, 281, 277, 263, 277, 251, 256, 258, 255, 269, 289, 265, 277, 281, 265, 271, 253, 266, 248, 282, 284, 348, 292, 264, 278, 288, 278, 280, 278, 278, 242, 267, 297, 270, 284, 272, 266, 298, 264, 252, 262, 262, 294, 269, 297, 246, 276, 239, 267, 278, 280, 260, 230, 274, 290, 273, 294, 262, 250, 284, 250, 264, 279, 263, 276, 282, 318, 248, 282, 284, 348, 292, 264, 278, 288, 278, 280, 278, 278, 242, 267, 297, 270, 284, 272, 266, 298, 264, 252, 262, 262, 294, 269, 297}
    builtins_test.go:165: z-score: 1.985714
    builtins_test.go:170: there is evidence that the generated IDs are not unordered (p < 0.05)
    panic.go:522: -- test log scope end --
--- FAIL: TestGenerateUniqueUnorderedIDOrder (0.10s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateUniqueUnorderedIDOrder.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/sem/builtins/builtins_test.go
+++ b/pkg/sql/sem/builtins/builtins_test.go
@@ -18,6 +18,7 @@ import (
 	"math/bits"
 	"math/rand"
 	"sort"
+	"strconv"
 	"testing"
 	"time"
 
@@ -125,11 +126,18 @@ func TestGenerateUniqueUnorderedIDOrder(t *testing.T) {
 
 	// We test our null hypothesis by running 100 trials and then performing
 	// a z-test (https://en.wikipedia.org/wiki/Z-test) for the observed mean
-	// number of swaps. We then compare it against the critical value of +-1.96,
-	// which corresponds to a confidence level of 95%.
+	// number of swaps.
 	const (
 		numTrials       = 100
 		numGensPerTrial = 101
+	)
+
+	// We then compare it against the critical value of +-2.58,
+	// which corresponds to a confidence level of 99%.
+	const (
+		confidenceLevel = 0.99
+		pValue          = 1 - confidenceLevel
+		criticalVal     = 2.58
 	)
 
 	// Run trials.
@@ -159,6 +167,8 @@ func TestGenerateUniqueUnorderedIDOrder(t *testing.T) {
 		}
 		swapCountsMean := float64(swapCountsSum) / float64(numTrials)
 
+		t.Logf("mean: %f", swapCountsMean)
+
 		stdErr := float64(distStdDev) / math.Sqrt(numTrials)
 		zScore = (swapCountsMean - float64(distMean)) / stdErr
 
@@ -166,8 +176,9 @@ func TestGenerateUniqueUnorderedIDOrder(t *testing.T) {
 	}
 
 	// Compare z-score to critical value.
-	if zScore < -1.96 || zScore > 1.96 {
-		t.Fatalf("there is evidence that the generated IDs are not unordered (p < 0.05)")
+	if zScore < -criticalVal || zScore > criticalVal {
+		t.Fatalf("there is evidence that the generated IDs are not unordered (p < %s)",
+			strconv.FormatFloat(pValue, 'f' /* fmt */, -1 /* prec */, 64 /* bitSize */))
 	}
 }
 

